File: programs/codechef/feb12_cookoff/comb.py
def fact(num):
	if num<=1: return 1
	return num*fact(num-1)

def nCr(n,r):
	if(n < r): return 0
	return fact(n) / (fact(r) * fact(n-r))

s = raw_input().split()
x = int(s[0])
n = int(s[1])
ans = 0
while n>0:
	n -= 1
	aval = raw_input()
	i=0; j=53;
	while(i<33):
		t=0
		for b in range(i,i+4):
			if(aval[b] == '0'): t += 1
		if(aval[j] == '0'): t += 1
		if(aval[j-1] == '0'): t += 1
		ans += nCr(t,x)
		i += 4
		j -= 2
print ans

File: programs/codechef/feb12_cookoff/comb.cpp
//Program question at: http://www.codechef.com/COOK19/problems/DAILY
#include <cstdio>
#include <iostream>
using namespace std;

int fact(int n) { return n<=1 ? 1 : n*fact(n-1); }

inline int nCr(int n, int r)
{
	if(n < r) return 0;
	return fact(n) / (fact(r) * fact(n-r));
}

int main()
{
	int x,n,ans=0; scanf("%d %d", &x, &n);
	while(n--)
	{
		char aval[55];
		aval[54] = '\0';
		cin >> aval;
		for(int i=0,j=53; i<33; i+=4,j-=2)
		{
			int t=0;
			for(int b=i;b<i+4;b++) if(aval[b] == '0') t++;
			if(aval[j] == '0') t++; if(aval[j-1] == '0') t++;
			ans += nCr(t,x);
		}
	}
	printf("%d\n",ans);
	return 0;
}
